src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java

```
@@ -0,0 +1,46 @@
+package mate.academy.service.impl;
+
+import java.util.Collections;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.dao.TicketDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.Ticket;
+import mate.academy.model.User;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private TicketDao ticketDao;
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket(movieSession, user);
+        ticketDao.add(ticket);
+        ShoppingCart shoppingCart = getByUser(user);
+        shoppingCart.getTickets().add(ticket);
+        shoppingCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        return shoppingCartDao.getByUser(user).get();
```

it's better to throw an exception in case if null in optional